runtime/port/linux/j9thread.c

```diff
@@ -0,0 +1,41 @@
+/*******************************************************************************
+ * Copyright IBM Corp. and others 2024
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] https://openjdk.org/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0 OR GPL-2.0-only WITH OpenJDK-assembly-exception-1.0
+ *******************************************************************************/
+
+#include <stdlib.h>
+#include <sys/resource.h>
+
+#include "portpriv.h"
+
+#define SEC_TO_NANO_CONVERSION_CONSTANT     (1000 * 1000 * 1000)
+#define MICRO_TO_NANO_CONVERSION_CONSTANT   1000
+
+int64_t
+j9thread_get_user_time(struct J9PortLibrary *portLibrary) {
```

Comment:
You need to add a new portlib function to omr. Then you can define a new type like `omrthread_thread_time_t` that returns both user and system time so you only need one call to get all the data.

Reply:
The new function will only be responsible for Linux right?

Reply:
Ideally, it would work the same on all platforms